src/lib.rs

```diff
@@ -120,17 +130,278 @@ fn footprint_latitude(e1: f64, mu: f64) -> f64 {
         + term4 * (8.0 * mu).sin()
 }
 
+const K0: f64 = 0.9996;
+const E: f64 = 0.00669438;
+
+#[cfg(feature = "std")]
+impl std::error::Error for WSG84ToLatLonError {
+    fn description(&self) -> &str {
+        match self {
+            WSG84ToLatLonError::EastingOutOfRange => {
+                "Easting out of range, must be between 100000 and 999999"
+            }
+            WSG84ToLatLonError::NorthingOutOfRange => {
+                "Northing out of range, must be between 0 and 10000000"
+            }
+            WSG84ToLatLonError::ZoneNumOutOfRange => {
+                "Zone num out of range, must be between 1 and 60"
+            }
+            WSG84ToLatLonError::ZoneLetterOutOfRange => {
+                "Zone letter out of range, must be between C and X"
+            }
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum WSG84ToLatLonError {
+    EastingOutOfRange,
+    NorthingOutOfRange,
+    ZoneNumOutOfRange,
+    ZoneLetterOutOfRange,
+}
+
+/// Converts a UTM coordinate to a latitude and longitude.
+/// zone_num can be obtain by calling lat_lon_to_zone_number
+/// zone_letter can be obtain by calling lat_to_zone_letter
+///
+/// # Example
```

Comment:
```suggestion
/// # Examples
```

The rust book recommends saying "Examples" even if there's just one: https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commonly-used-sections.